src/find/matchers/mod.rs

```diff
@@ -0,0 +1,732 @@
+mod logical_matchers;
```

Comment:
Maybe considering make some of these modules as public (also in other modules) so we can actually call the functionality as a library? But definitely can wait.